pkg/webhook/creator/mutatingwebhookcfg.go

```diff
@@ -0,0 +1,58 @@
+//
+// Copyright (c) 2019-2020 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+package creator
+
+import (
+	"github.com/che-incubator/che-workspace-operator/pkg/webhook/server"
+	"k8s.io/api/admissionregistration/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	mutateWebhookCfgName       = "mutate-workspace-admission-hooks"
+	mutateWebhookPath          = "/mutate-workspaces"
+	mutateWebhookFailurePolicy = v1beta1.Fail
+)
+
+func buildMutateWebhookCfg() *v1beta1.MutatingWebhookConfiguration {
```

Comment:
On created webhook configs you have to add the owner reference
If you correctly set the owner reference on the created configuration objects, there will be no cleaning problem as soon as you uninstall / reinstall the controller deployment.
This is already done for the metrics service (created by the operator-sdk), and was implemented the same way to managed the service for the embedded plugin registry:
https://github.com/sleshchenko/che-workspace-crd-operator/blob/a99704e63b3d49446a094d2ef6460e383974ac22/pkg/controller/registry/embedded_registry.go#L135

Additionally, as soon as you have set the owner ref on the webhook configuration you create / update, you would even be able, at controller start (when adding / updating the webhook configs), to detect if you have owned webhook configs with different names than the ones you have to create / update, and remove them.